### PR TITLE
[FIX] 11.0 stock: fix slow inventory valuation

### PR DESCRIPTION
**Pull request by mpanarin**
Description of the issue/feature this PR addresses:

When starting an inventory with more that 300 lines
every change in the lines takes ~5-6 seconds, which grows
with the amount of lines in inventory. This is due to this
relation that triggers a lot of unnecessary recalculations.

As this field is actually never used in code -> I think it should be dropped 
for the sake of performance

Current behavior before PR:

When large inventory is started (for "All products" for example) every change to the inventory 
lines take a lot of time, which increases exponentially with the amount of lines.

Desired behavior after PR is merged:

It should be hella fast

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
Sorry but we can't delete a field in a stable version, this is against the [stable policy](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean).
We could drop it in master if not needed.
For stable removing from the view is not enough?
Are you sure it is this field that is slow? A related (moreover with an index on the relation) is usually not very slow, especially if the value does not change.

**Comment by pedrobaeza** [comment]
I think removing it from the view makes it to not be loaded.

**Comment by gdgellatly** [comment]
@mpanarin this issue has dogged many of us for years.  Odoo even created special views of just lines trying to fix it.  I would love to see some performance stats here because if it really is that simple it would be brilliant.

**Comment by mpanarin** [comment]
@pedrobaeza @gdgellatly yes, removing it just from views fixes the issue as well.
Change of Real Quantity in the inventory with ~650 items:
Before dropping the field from view - 5-8 seconds
After dropping the field - less than half a second.

@mart-e Now more on the issue itself:
1. Odoo treats related fields as an onchange on both sides. Due to the nature of related fields you can change the value of from both related records so here: https://github.com/odoo/odoo/blob/11.0/odoo/fields.py#L676
it creates inverse dependencies for the lines.
result of `resolve_deps` for model `stock.inventory`:
```
[(stock.inventory(), stock.inventory.product_id, []),
 (stock.inventory(), stock.inventory.line_ids, []),
 (stock.inventory.line(), stock.inventory.line.product_qty, ['line_ids']),
 (stock.inventory.line(), stock.inventory.line.inventory_id, ['line_ids'])]
```
`product_qty` is a dependency as it is used in computation of `total_qty` field.
`inventory_id` as line has a related field.
2. This all leads to an issue itself - 
https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L5069
https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L5104
dotnames here is:
```
['move_ids.product_id',
 'move_ids.location_id',
 'filter',
 'move_ids.product_uom_qty',
 'line_ids.product_uom_id',
 'move_ids.scrapped',
 'line_ids.partner_id',
 'move_ids.product_uom',
 'category_id',
 'company_id',
 'line_ids.state',
 'product_id',
 'move_ids.date_expected',
 'location_id',
 'scan_location_id',
 'state',
 'date',
 'line_ids',
 'partner_id',
 'line_ids.theoretical_qty',
 'name',
 'line_ids.product_barcode',
 'line_ids.description_picking',
 'line_ids.location_id',
 '_barcode_scanned',
 'exhausted',
 'line_ids.product_id',
 'move_ids.location_dest_id',
 'accounting_date',
 'move_ids.state',
 'lot_id',
 'move_ids.picking_id',
 'move_ids.create_date',
 'line_ids.product_qty',
 'line_ids.package_id',
 'line_ids.prod_lot_id',
 'move_ids',
 'package_id']
```
mapping this amount of fields on 600+ lines obviously takes a ton of time.
So the issue is not in computation directly, nor in index or the related field, but in mapping this amount of fields because of this relation.

Dropping the field is obviously a cure to the symptom, not the issue itself. But as this field isn't used anywhere its a good temporary solution.

Basically, we will get the same issue with any related fields and a record with a huge amount of lines

**Review by nhomar — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@mart-e any update on this?

**Comment by pedrobaeza** [comment]
Martin is on holidays. Maybe @sle-odoo can tell us.

**Comment by gdgellatly** [comment]
Just an update. I used this patch in v12 (well I removed the field from the view) a few weeks ago for an unbearably slow annual stocktake say 50 seperate stocktakes of 1000 lines each, some more some less. Worked perfect. No stats, I don't enter them, but everyone happy.

**Comment by mpanarin** [comment]
Well, no idea why it wasn't merged :\
@pedrobaeza @mart-e

**Comment by C3POdoo** [comment]
Dear @mpanarin,

Thank you for your contribution but the version 11.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._
